File: src/main/java/com/yerti/core/utils/ProgressBar.java
package com.yerti.core.utils;

import org.bukkit.ChatColor;

public class ProgressBar {

    private String symbol;
    private int percent;
    private int max;
    private ChatColor empty;
    private ChatColor filled;
    private int barAmount;

    public ProgressBar(String symbol, int percent, int max, int barAmount, ChatColor empty, ChatColor filled) {
        this.symbol = symbol;
        this.max = max;
        this.percent = percent;
        this.empty = empty;
        this.filled = filled;
        this.barAmount = barAmount;
    }

    public String getBar() {
        double neededPercent = (double) percent / max;
        int bars = (int) (barAmount * (neededPercent / 100.));

        StringBuilder builder = new StringBuilder();

        for (int i = 1; i <= barAmount; i++) {
            if (i <= bars) {
                builder.append(filled).append(symbol);
            } else {
                builder.append(empty).append(symbol);
            }
        }

        return builder.toString();


    }

    public void updatePercent(int percent) {
        this.percent = percent;
    }


}

File: src/main/java/com/yerti/core/utils/ProgressBar.kt
package com.yerti.core.utils

import org.bukkit.ChatColor

class ProgressBar(private val symbol: String, private var percent: Int, private val max: Int, private val barAmount: Int, private val empty: ChatColor, private val filled: ChatColor) {
    val bar: String
        get() {
            val neededPercent = percent.toDouble() / max
            val bars = (barAmount * (neededPercent / 100.0)).toInt()
            val builder = StringBuilder()
            for (i in 1..barAmount) {
                if (i <= bars) {
                    builder.append(filled).append(symbol)
                } else {
                    builder.append(empty).append(symbol)
                }
            }
            return builder.toString()
        }

    fun updatePercent(percent: Int) {
        this.percent = percent
    }

}